File: lib/java/src/main/java/uapi/internal/schema/FindMatchingSchemaKey.java
package uapi.internal.schema;

import java.util.Objects;
import java.util.Set;

public class FindMatchingSchemaKey {
    public static String findMatchingSchemaKey(Set<String> schemaKeys, String schemaKey) {
        for (final var k : schemaKeys) {
            if (schemaKey.startsWith("info.") && k.startsWith("info.")) {
                return k;
            }
            if (Objects.equals(k, schemaKey)) {
                return k;
            }
        }
        return null;
    }
}

File: lib/py/uapi/internal/schema/FindMatchingSchemaKey.py
def find_matching_schema_key(schema_keys: set[str], schema_key: str) -> str | None:
    for k in schema_keys:
        if schema_key.startswith("info.") and k.startswith("info."):
            return k
        if k == schema_key:
            return k
    return None
